src/objectid.ts

```diff
@@ -275,9 +285,16 @@ export class ObjectId extends BSONValue {
    * @param time - an integer number representing a number of seconds.
    */
   static createFromTime(time: number): ObjectId {
-    const buffer = ByteUtils.fromNumberArray([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+    const buffer = ByteUtils.allocate(12);
+    for (let i = 11; i >= 4; i--) buffer[i] = 0;
     // Encode time into first 4 bytes
-    BSONDataView.fromUint8Array(buffer).setUint32(0, time, false);
+    buffer[3] = time;
+    time >>>= 8;
+    buffer[2] = time;
+    time >>>= 8;
+    buffer[1] = time;
+    time >>>= 8;
+    buffer[0] = time;
```

Comment:
If we write them carefully no I don't think it should introduce overhead, and I agree it could be more organized. I migrated the code to function calls on another branch and I'll share with you a comparison soon. (I already determined it is an improvement over main, but I wonder how it compares against this branch)

Reply:
[benchmarks](https://docs.google.com/spreadsheets/d/1gRaiQGJYDnxh4TRn6HNm0KxBQkyEZBAmKEeGEPXJpuI/edit#gid=2086700153)

I think inlining the functions doesn't introduce much overhead so we can clean things up, I'll open a new PR against this one with the changes.